testtools/files.go

```diff
@@ -398,3 +398,9 @@ func updateExpectedConfig(t *testing.T, expected string, actual string, srcTestD
 		}
 	}
 }
+
+// redactWorkspacePath replaces workspace path with a constant to make the test
+// output reproducible.
+func redactWorkspacePath(s, wsPath string) string {
+	return strings.ReplaceAll(s, wsPath, "%WORKSPACEPATH%")
```

Comment:
I think a bit more is needed to make this work in Windows

```
Executing tests from //tests:fix_mode_strict
-----------------------------------------------------------------------------
--- FAIL: TestFullGeneration (0.04s)
    --- FAIL: TestFullGeneration/fix_mode_strict (0.04s)
        files.go:187: expected gazelle stderr: gazelle: <WORKSPACE>/BUILD.bazel:3:13: syntax error near visibility
            gazelle: Exit as strict mode is on
            
            got: gazelle: <WORKSPACE>\BUILD.bazel:3:13: syntax error near visibility
            gazelle: Exit as strict mode is on
            
        files.go:306: 
            =====================================================================================
            Run UPDATE_SNAPSHOTS=true bazel run //tests:fix_mode_strict to update BUILD.out and expected{Stdout,Stderr,ExitCode}.txt files.
            =====================================================================================
FAIL
```

From [here](https://buildkite.com/bazel/bazel-gazelle/builds/1752#9a5fa00c-43f0-4e96-a5ce-68af6d78257e)

I think the difference is the direction of the `/` character. I wanted to quick flag flag this in case you've got a clever strategy for fixing this.

Reply:
Actually, now that I read your comment it seems something you're aware of. Let's merge this and talk it over in the CL where the tests is actually failing.